cmd/frontend/graphqlbackend/graphqlutil/connection_resolver.go

```diff
@@ -0,0 +1,271 @@
+package graphqlutil
+
+import (
+	"context"
+	"sync"
+
+	"github.com/graph-gophers/graphql-go"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+const DEFAULT_MAX_PAGE_SIZE = 100
+
+type ConnectionResolver[N ConnectionNode] struct {
+	store   ConnectionResolverStore[N]
+	args    *ConnectionResolverArgs
+	options *ConnectionResolverOptions
+	data    connectionData[N]
+	once    resolveOnce
+}
+
+type ConnectionNode interface {
+	ID() graphql.ID
+}
+
+type ConnectionResolverStore[N ConnectionNode] interface {
+	// ComputeTotal is required to fetch the total number of items.
```

Comment:
```suggestion
	// ComputeTotal returns the total count of all the items in the connection, independent of pagination arguments.
```

See: https://go.dev/blog/godoc for how to write docstrings in Go.

Same goes for the other methods here.